raiden/network/transport/matrix/transport.py

```diff
@@ -833,15 +813,61 @@ def _initialize_broadcast_rooms(self) -> None:
         msg = "To join the broadcast rooms the Matrix client to be properly authenticated."
         assert self._user_id, msg
 
-        for suffix in self._config.broadcast_rooms:
-            room_name = make_room_alias(self.chain_id, suffix)
-            broadcast_room_alias = f"#{room_name}:{self._server_name}"
+        pool = Pool(size=10)
 
-            self.log.debug("Joining broadcast room", broadcast_room_alias=broadcast_room_alias)
-            self._broadcast_rooms[room_name] = join_broadcast_room(
-                client=self._client, broadcast_room_alias=broadcast_room_alias
+        def _join_broadcast_room(transport: MatrixTransport, room_name: str) -> None:
+            broadcast_room_alias = f"#{room_name}:{transport._server_name}"
+            transport.log.debug(
+                "Joining broadcast room", broadcast_room_alias=broadcast_room_alias
+            )
+            transport._broadcast_rooms[room_name] = join_broadcast_room(
+                client=transport._client, broadcast_room_alias=broadcast_room_alias
             )
 
+        for suffix in self._config.broadcast_rooms:
+            alias_prefix = make_room_alias(self.chain_id, suffix)
+
+            if alias_prefix not in self._broadcast_rooms:
+                pool.apply_async(_join_broadcast_room, args=(self, alias_prefix))
+
+        pool.join(raise_error=True)
+
+    def _initialize_sync(self) -> None:
+        msg = "_initialize_sync requires the GMatrixClient to be properly authenticated."
+        assert self._user_id, msg
+
+        msg = "The sync thread must not be started twice"
+        assert self._client.sync_worker is None, msg
+        assert self._client.message_worker is None, msg
+
+        msg = (
+            "The node must have joined the broadcast rooms before starting the "
+            "sync thread, since that is necessary to properly generate the "
+            "filters."
+        )
+        assert self._broadcast_rooms, msg
```

Comment:
is this right? do we join the broadcast rooms if the node is not configured to use the ms and pfs?

Reply:
We will always join at least the `discovery` room.